packages/compiler/core/checker.ts

```diff
@@ -2273,6 +2340,83 @@ export function createChecker(program: Program): Checker {
     properties.set(newProp.name, newProp);
   }
 
+  function bindAllMembers(node: Node) {
+    if (node.symbol) {
+      bindMembers(node, node.symbol);
+    }
+    visitChildren(node, (child) => {
+      bindAllMembers(child);
+    });
+  }
+
+  function bindMembers(node: Node, containerSym: Sym) {
+    let containerMembers: Mutable<SymbolTable>;
+
+    switch (node.kind) {
+      case SyntaxKind.ModelStatement:
+        if (node.extends && node.extends.kind === SyntaxKind.TypeReference) {
+          resolveAndCopyMembers(node.extends);
+        }
+        if (node.is && node.is.kind === SyntaxKind.TypeReference) {
+          resolveAndCopyMembers(node.is);
+        }
+        for (const prop of node.properties) {
+          if (prop.kind === SyntaxKind.ModelSpreadProperty) {
+            resolveAndCopyMembers(prop.target);
+          } else {
+            const name = prop.id.kind === SyntaxKind.Identifier ? prop.id.sv : prop.id.value;
+            bindMember(name, prop, SymbolFlags.ModelProperty);
+          }
+        }
+        break;
+      case SyntaxKind.EnumStatement:
+        for (const member of node.members.values()) {
+          if (member.kind === SyntaxKind.EnumSpreadMember) {
+            resolveAndCopyMembers(member.target);
+          } else {
+            const name = member.id.kind === SyntaxKind.Identifier ? member.id.sv : member.id.value;
+            bindMember(name, member, SymbolFlags.EnumMember);
+          }
+        }
+        break;
+      case SyntaxKind.InterfaceStatement:
+        for (const member of node.operations.values()) {
+          bindMember(member.id.sv, member, SymbolFlags.InterfaceMember);
+        }
+        break;
+      case SyntaxKind.UnionStatement:
+        for (const variant of node.options.values()) {
+          const name = variant.id.kind === SyntaxKind.Identifier ? variant.id.sv : variant.id.value;
+          bindMember(name, variant, SymbolFlags.UnionVariant);
+        }
+        break;
+    }
+
+    function resolveAndCopyMembers(node: TypeReferenceNode) {
+      let ref = resolveTypeReferenceSym(node, undefined);
```

Comment:
I would just like to thank `evalCadlScript` for being deleted permanently and making such resolution at this point safe to do.